### PR TITLE
fix(dev): Enable line buffering for stderr

### PR DESCRIPTION
**Pull request by wmak**
- Even with the `PYTHONUNBUFFERED` envvar, in python3 stderr is still being
  buffered which means that errors don't show up until the server is killed.
  This enables line buffering so that even in py3.6 these
  errors show up in our console immediately
- In py3.9 stderr is line buffered by default so this change will no
  longer be required, but until then we'll need it.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
It'd be better to put something like this inside of https://github.com/getsentry/sentry/blob/master/src/sentry/runner/initializer.py#L275

**Comment by mattrobenolt** [comment]
And arguably, just use `settings.DEBUG` instead. It fits more in line with `DEBUG` rather than `IS_DEV` imo.

**Comment by iProgramStuff** [comment]
This is gold, Will. Thanks for figuring this out....I'm sure a lot of people will appreciate this, including me (I already use this in my own `server.py` thanks to you)
Screen Shot 2021-01-06 at 10 19 49 AM

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -69,6 +69,10 @@ def env(key, default="", type=None):
 IS_DEV = ENVIRONMENT == "development"
 
 DEBUG = IS_DEV
+if IS_DEV:
+    # Enable line buffering for stderr, TODO(py3.9) can be removed after py3.9, see bpo-13601
+    sys.stderr = os.fdopen(sys.stderr.fileno(), "w", 1)
```
Comment (iProgramStuff) [inline src/sentry/conf/server.py:74]: This works perfectly from what I can tell. I agree with just using `DEBUG` directly...and don't have a strong opinion on where this snippet lands so I'll defer to Matt on that

**Review by iProgramStuff — APPROVED** [review]
This looks good to me, but maybe @mattrobenolt should take another look too?

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**